src/model/UserphraseModel.cpp

```diff
@@ -238,11 +238,11 @@ const Userphrase *UserphraseModel::getUserphrase(const QModelIndex& idx)
 
 void UserphraseModel::undo()
 {
-    if (!userphrase_tmp_.empty()) {
```

Comment:
You can safely rewrite the method by operating "shadow" without the need of helper functions.